src/xdg.c

```diff
@@ -114,14 +121,26 @@ handle_destroy(struct wl_listener *listener, void *data)
 {
 	struct view *view = wl_container_of(listener, view, destroy);
 	assert(view->type == LAB_XDG_SHELL_VIEW);
+	struct xdg_toplevel_view *xdg_toplevel_view =
+		xdg_toplevel_view_from_view(view);
+
+	xdg_toplevel_view->xdg_surface->data = NULL;
+	xdg_toplevel_view->xdg_surface = NULL;
+
+	wl_list_remove(&view->map.link);
+	wl_list_remove(&view->unmap.link);
+	wl_list_remove(&view->request_move.link);
+	wl_list_remove(&view->request_resize.link);
+	wl_list_remove(&view->request_minimize.link);
+	wl_list_remove(&view->request_maximize.link);
+	wl_list_remove(&view->request_fullscreen.link);
+	wl_list_remove(&view->set_title.link);
```

Comment:
We could also remove the common listeners in `view_destroy()` instead.

Reply:
Done